### PR TITLE
IS-82 PrincipalSelection updates

### PR DESCRIPTION
**Pull request by martin-lindstrom**
The deployment profile and the BankID profile were updated with recommendations and requirements for PrincipalSelection use.

### CODE REVIEW

File: ELN-0602 - Deployment Profile for the Swedish eID Framework.md [hunk 1]
```diff
@@ -1056,6 +1081,8 @@ following requirements:
     also be indicated in the Signature Service metadata record using the
     `AuthnRequestsSigned` attribute (see [section 2.1.4](#signature-service)).
 
+If the receiving Identity Provider has declared that it wishes to receive certain attributes known about the user that is performing the signature operation (see [section 2.1.3](#identity-providers)), and the Signature Service has received any of the declared attributes in the `Signer` element of the `SignRequest` (\[[EidDSS](#eiddss)\]), the Signature Service SHOULD include those attributes in a `<psc:PrincipalSelection>` extension of the authentication request. See also [section 5.3.3](#principal-selection).
+
 It is RECOMMENDED that the `<saml2p:Scoping>` element containing a `<saml2p:RequesterID>` element holding the entityID of the Service Requestor is included in `<saml2p:AuthnRequest>` messages generated by a Signature Service.
```
Comment (Razumain) [inline ELN-0602 - Deployment Profile for the Swedish eID Framework.md]: Suggested wording:

If the receiving Identity Provider has declared that it wishes to receive principal selection information about the signer (see [section 5.3.3](#principal-selection)), and the Signature Service has received any of the requested attribute values in the `Signer` element of the `SignRequest` (\[[EidDSS](#eiddss)\]), the Signature Service SHOULD include those attribute values in a `<psc:PrincipalSelection>` extension of the authentication request.

File: ELN-0614 - Principal Selection in SAML Authentication Requests.md [hunk 2]
```diff
@@ -120,30 +132,86 @@ The following schema fragment defines the `<MatchValueType>` complex type:
...
+The following schema fragment defines the `<RequestedPrincipalSelection>` element:
+
+```
+<element name="RequestedPrincipalSelection" type="psc:RequestedPrincipalSelectionType" />
+<complexType name="RequestedPrincipalSelectionType">
+  <complexContent>
+    <extension base="psc:PrincipalSelectionType" />
+  </complexContent>
+</complexType>
+```
```
Comment (Razumain) [inline ELN-0614 - Principal Selection in SAML Authentication Requests.md]: xml prefix declarations are missing. Update to:

  <xs:element name="RequestedPrincipalSelection" type="psc:RequestedPrincipalSelectionType" />
  <xs:complexType name="RequestedPrincipalSelectionType">
    <xs:complexContent>
      <xs:extension base="psc:PrincipalSelectionType" />
    </xs:complexContent>
  </xs:complexType>

File: ELN-0614 - Principal Selection in SAML Authentication Requests.md [hunk 3]
```diff
@@ -120,30 +132,86 @@ The following schema fragment defines the `<MatchValueType>` complex type:
...
        <xs:attribute name="NameFormat" type="xs:anyURI" 
             default="urn:oasis:names:tc:SAML:2.0:attrname-format:uri" />
        <xs:attribute name="Name" type="xs:string" use="required" />
        <xs:anyAttribute namespace="##any" />
      </xs:extension>
    </xs:simpleContent>
  </xs:complexType>
  
</xs:schema>
 ```
```
Comment (Razumain) [inline ELN-0614 - Principal Selection in SAML Authentication Requests.md]: xml prefix missing for RequestedPrincipalSelection. See above

File: ELN-0602 - Deployment Profile for the Swedish eID Framework.md [hunk 4]
```diff
@@ -562,6 +576,17 @@ proxy the request to the Norwegian eIDAS Proxy Service for authentication.*
...
+#### 5.3.3. Principal Selection
+
+The specification "Principal Selection in SAML Authentication Requests", \[[PrincipalSel](#principalsel)\], defines the `<psc:PrincipalSelection>` extension that enables a requestor to inform the Identity Provider about known attributes for the principal that is about to be authenticated. A typical case is when an already logged in user is sent to the Identity Provider for re-authentication, and it is undesirable that the user is prompted for its user ID.
+
+An Identity Provider that wishes to receive the `<psc:PrincipalSelection>` extension SHOULD advertise this in its metadata according to [section 2.1.3](#identity-providers).
+
+A Service Provider that sends an authentication request  to an Identity Provider that has declared that it wishes to receive certain attribute values MAY include the `<psc:PrincipalSelection>` extension in the `<saml2p:AuthnRequest>` message, provided that the Service Provider has knowledge about this information.
+
+A Service Provider that is a Signature Service SHOULD include the extension for the case described above.
+
```
[the remark this answers is not in the document]
Comment (martin-lindstrom) [reply ELN-0602 - Deployment Profile for the Swedish eID Framework.md]: We refer to the case above, and that text states that the IdP has requested this information.

**Pull request merged**